daemon/src/file_asset_source.rs

```diff
@@ -833,21 +831,10 @@ impl FileAssetSource {
                         .import_state
                         .import_hash()
                         .expect("path changed but no import hash present");
-                    // TODO set error state for unresolved path references?
-                    // this code strips out path references for now, but it will probably crash and burn when loading
```

Comment:
This is the one thing I'm not 100% sure of.  Didn't experience any failures as this comment would lead me to believe...

Reply:
I guess this refers to the silent failure in the original bug, if the path is unresolved it never seems to reach this code anyway.

Reply:
Sorry about the confusion. This code is in the daemon and concerns resolving path references to UUIDs in assets during *Import*, not loading.

Here's an example of how this feature is used:
https://github.com/amethyst/atelier-assets/blob/master/examples/handle_integration/assets/custom_asset.ron#L5
https://github.com/amethyst/atelier-assets/blob/master/examples/handle_integration/src/custom_asset.rs#L12

The original bug concerns IndirectIdentifiers never re-resolving, i.e. when loading an asset by path *at runtime, in-engine*.
This code path should rerun when the relevant path has changed: https://github.com/amethyst/atelier-assets/blob/master/loader/src/loader.rs#L830
Here's where the state of the resolve operations live: https://github.com/amethyst/atelier-assets/blob/master/loader/src/loader.rs#L109

Reply:
I had another branch where I was messing with the loader part, let me see if I can reconcile and revert the possibly breaking change in this comment.